src/test/groovy/com/github/rodionovsasha/commentservice/services/TopicServiceTest.groovy

```diff
@@ -0,0 +1,211 @@
+
+        then:
+        thrown(UserNotFoundException)
+    }
+
+    def "archive makes a topic archived"() {
+        given:
+        !topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, HOMER_ID)
+
+        then:
+        topicService.getById(TOPIC_ID).archived
+    }
+
+    def "archive throws when user is inactive"() {
+        given:
+        !topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, 2)
+
+        then:
+        !topicService.getById(TOPIC_ID).archived
+
+        and:
+        thrown(InactiveUserException)
+    }
+
+    def "archive throws when user not found"() {
+        given:
+        !topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, NOT_EXISTING_USER_ID)
+
+        then:
+        !topicService.getById(TOPIC_ID).archived
+
+        and:
+        thrown(UserNotFoundException)
+    }
+
+    def "archive throws when user is not topic owner"() {
+        given:
+        def TOPIC_ID = 3
+        !topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, HOMER_ID)
+
+        then:
+        !topicService.getById(TOPIC_ID).archived
+
+        and:
+        def e = thrown(TopicAccessException)
+        e.message == "Non-topic owner with id '1' is trying to archive the topic"
+    }
+
+    def "archive does not change already archived topic"() {
+        given:
+        def TOPIC_ID = 7
+        topicService.getById(TOPIC_ID).archived
+
+        when:
+        topicService.archive(TOPIC_ID, 1)
+
+        then:
+        topicService.getById(TOPIC_ID).archived
+    }
+
+    def "listForUser returns all topics for user ASC sorted"() {
+        when:
+        def topics = topicService.listForUser(HOMER_ID, new Sort(Sort.Direction.ASC, "id"))
+
+        then:
+        with(topics) {
+            size() == 5
+            id == [1, 2, 5, 6, 7]
+        }
+    }
+
+    def "listForUser returns all topics for user DESC sorted"() {
+        when:
+        def topics = topicService.listForUser(HOMER_ID, new Sort(Sort.Direction.DESC, "id"))
+
+        then:
+        with(topics) {
+            size() == 5
+            id == [7, 6, 5, 2, 1]
+        }
+    }
+
+    def "listForUser returns an empty list when user does not have own topics"() {
+        when:
+        def topics = topicService.listForUser(3, new Sort(Sort.Direction.ASC, "id"))
+
+        then:
+        topics.size() == 0
+    }
+
+    def "listForUser throws when user is inactive"() {
+        when:
+        topicService.listForUser(2, null)
+
+        then:
+        thrown(InactiveUserException)
+    }
+
+    def "listForUser throws when user not found"() {
+        when:
+        topicService.listForUser(NOT_EXISTING_USER_ID, null)
+
+        then:
+        thrown(UserNotFoundException)
+    }
+
+    def "search returns topics with a fragment in title"() {
```

Comment:
Let's check sorting `get(0).date > get(1).date` in a separate test.
Also, we need tests for:
* Case when nothing was found
* Case when the limit(the second parameter of `search("Flanders", 10)`) actually limits the result size
* Is search case sensitive or not? (I assume it should be case insensitive)